test_runner/src/main/kotlin/ftl/api/Artifacts.kt

```diff
@@ -0,0 +1,34 @@
+package ftl.api
+
+import ftl.adapter.GcFetchArtifacts
+
+val fetchArtifacts: Artifacts.Fetch get() = GcFetchArtifacts
+
+object Artifacts {
+
+    data class Identity(
+        val gcsPathWithoutRootBucket: String,
+        val gcsRootBucket: String,
+        val regex: List<Regex>,
+        val blobPath: String, // todo not needed?
+        val downloadPath: DownloadPath,
+        val matrixId: String
```

Comment:
Added `matrixId` to persists info about matrix from which files are downloaded. It is used to update `download` flag once all files are fetched. Let me know if this violates any assumptions.
https://github.com/Flank/flank/pull/1875/files#diff-f91686b92f886f88f9b047c859da24c40bc6ec3ee8686ef5eec4bd5259f91de2R45

@jan-gogo